src/tracing/node_trace_buffer.cc

```diff
@@ -59,7 +59,13 @@ void InternalTraceBuffer::Flush(bool blocking) {
       for (size_t i = 0; i < total_chunks_; ++i) {
         auto& chunk = chunks_[i];
         for (size_t j = 0; j < chunk->size(); ++j) {
-          agent_->AppendTraceEvent(chunk->GetEventAt(j));
+          TraceObject* trace_event = chunk->GetEventAt(j);
```

Comment:
Why not `auto`?

Reply:
IMHO it's time for that to change ([ES.11](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto)).
Beside using the CCG to not bikeshed, for me as a reader `TraceObject*` is as opaque as `auto`, that why we have IDEs.